src/py-opentimelineio/opentimelineio/adapters/otioz.py

```diff
@@ -0,0 +1,140 @@
+locally referenced or provide missing references
+
+Can also extract the content.otio file from an otioz bundle for processing.
+"""
+
+import os
+import copy
+import zipfile
+
+from .. import (
+    exceptions,
+    schema,
+)
+
+from . import (
+    file_bundle_utils as utils,
+    otio_json
+)
+
+
+def read_from_file(filepath, extract_to_directory=None):
+    if not zipfile.is_zipfile(filepath):
+        raise exceptions.OTIOError("Not a zipfile: {}".format(filepath))
+
+    if extract_to_directory:
+        output_media_directory = os.path.join(
+            extract_to_directory,
+            utils.BUNDLE_DIR_NAME
+        )
+
+        if not os.path.exists(extract_to_directory):
+            raise exceptions.OTIOError(
+                "Directory '{}' does not exist, cannot unpack otioz "
+                "there.".format(extract_to_directory)
+            )
+
+        if os.path.exists(output_media_directory):
+            raise exceptions.OTIOError(
+                "Error: '{}' already exists on disk, cannot overwrite while "
+                " unpacking OTIOZ file '{}'.".format(
+                    output_media_directory,
+                    filepath
+                )
+
+            )
+
+    with zipfile.ZipFile(filepath, 'r') as zi:
+        result = otio_json.read_from_string(zi.read(utils.BUNDLE_PLAYLIST_PATH))
+
+        if extract_to_directory:
+            zi.extractall(extract_to_directory)
+
+    return result
+
+
+def write_to_file(
+    input_otio,
+    filepath,
+    media_policy=utils.MediaReferencePolicy.ErrorIfNotFile,
+    dryrun=False
+):
+    input_otio = copy.deepcopy(input_otio)
+
+    manifest = utils._file_bundle_manifest(
+        input_otio,
+        filepath,
+        media_policy,
+        "OTIOZ"
+    )
+
+    # dryrun reports the total size of files
+    if dryrun:
+        fsize = 0
+        for fn in manifest:
+            fsize += os.path.getsize(fn)
+        return fsize
+
+    fmapping = {}
+
+    # gather the files up in the staging_dir
+    for fn in manifest:
+        target = os.path.join(utils.BUNDLE_DIR_NAME, os.path.basename(fn))
+        fmapping[fn] = target
+
+    # relink the media reference
+    for cl in input_otio.each_clip():
+        if media_policy == utils.MediaReferencePolicy.AllMissing:
+            cl.media_reference = schema.MissingReference()
+            continue
+
+        try:
+            source_fpath = cl.media_reference.target_url
+        except AttributeError:
+            continue
+
+        cl.media_reference.target_url = "file://{}".format(
+            fmapping[source_fpath.split("file://")[1]]
+        )
+
+    # write the otioz file to the temp directory
+    otio_str = otio_json.write_to_string(input_otio)
+
+    with zipfile.ZipFile(filepath, mode='w') as target:
+        # write the media
+        for src, dst in fmapping.items():
+            target.write(src, dst)
+
+        # write the OTIO
+        target.writestr(utils.BUNDLE_PLAYLIST_PATH, otio_str)
```

Comment:
We should pass in `compress_type=ZIP_LZMA` for the json part since its going to be parsed anyway.  Then we get the benefits of the zip file format for people concerned with large otio file size.

Reply:
turns out `ZIP_LZMA` is python 3 only, so we're using regular zip compression for now.